manheim_c7n_tools/dryrun_diff.py

```diff
@@ -100,6 +115,100 @@ def _find_changed_policies(self, git_dir=None, diff_against='master'):
             pnames.append(m.group(1))
         return pnames
 
+    def _make_diff_report(self, dryrun):
+        """
+        Return a HTML report breaking down the differences between the dryrun
+        (this branch) and the last run of each policy on master.
+
+        Requires a jinja template located at `./reporting-template/report.j2`
+        within the same directory where the dryrun-diff step has been run.
+
+        :param dryrun: dryrun policy resource information
+        :type dryrun: dict
+        :return: html report
+        :rtype: str
+        """
+        all_policies = list(
+            set(dryrun.keys()) | set(self._live_results.keys())
+        )
+        entries = {}
+        if not all_policies:
+            logger.info('no policies found - skipping diff report')
+            return ''
+        t_loader = FileSystemLoader(searchpath="./reporting-template/")
+        t_env = Environment(loader=t_loader)
+        t_file = "report.j2"
+        try:
+            tmpl = t_env.get_template(t_file)
+        except TemplateNotFound:
+            logger.info('unable to find a template - skipping diff report')
+            return ''
+
+        for policy in all_policies:
+            policy_splits = policy.split('-', 1)
+            if len(policy_splits) <= 1:
+                logger.warning(
+                    "skipping policy: \
+                        %s from report - it is not prefixed correctly",
+                    policy)
+                continue
+            logger.info(
+                'getting policy: %s resource type from live run', policy)
+            resource_type = self._live_results.get(policy, {}).get(
+                self.RESOURCE_TYPE_KEY, self.UNKNOWN_RESOURCE_TYPE)
+            if resource_type == self.UNKNOWN_RESOURCE_TYPE:
+                logger.warning(
+                    'unable to get policy: %s resource type from live run', policy)
+                logger.info(
+                    'getting policy: %s resource type from dryrun', policy)
+                resource_type = dryrun.get(policy, {}).get(
+                    self.RESOURCE_TYPE_KEY, self.UNKNOWN_RESOURCE_TYPE)
+                if resource_type == self.UNKNOWN_RESOURCE_TYPE:
+                    logger.error('unable to get policy: %s resource type', policy)
+                    return ''
```

Comment:
What if we change the resource the policy affects, but not it's name - this might be an issue. I'll work on this.